src/bin/cargo.rs

```diff
@@ -91,6 +92,13 @@ macro_rules! each_subcommand{ ($mac:ident) => ({
   on this top-level information.
 */
 fn execute(flags: Flags, config: &Config) -> CliResult<Option<()>> {
+    let mut fl = CargoLock::new(config.home().join(".global-lock"),
+                                try!(CargoLock::lock_kind(config)));
+
+    try!(fl.lock().chain_error(|| {
+        human("Failed to obtain global cargo lock")
```

Comment:
> I think that we'll want to improve this error message slightly, for example how can this come up?
Good idea. Also to check for IOErrors specifically. These shouldn't happen though, and will be provided as cause as FileLock errors are declared *human* right now.